### PR TITLE
Fix overlapping line comments issue in TexlLexer

### PR DESCRIPTION
**Pull request by nguhoa**
Fix issue: https://github.com/microsoft/Power-Fx/issues/1444

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Core/Lexer/TexlLexer.cs [hunk 1]
```diff
@@ -899,6 +899,7 @@ public enum LexerMode
             private readonly bool _disableReservedKeywords;
 
             private int _currentTokenPos; // The start of the current token.
+            private bool _previousHasEndLineComment = false; // True/False if the last previous commment already process end line character
```
[the remark this answers is not in the document]
Comment (nguhoa) [reply src/libraries/Microsoft.PowerFx.Core/Lexer/TexlLexer.cs]: Thanks for suggestion. After rethink about this and found that I just need to store last comment end index and the check of preceding will happen until that last index. add more tests to confirm this as well. Please review again

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by akshar2401 — APPROVED** [review]

File: src/libraries/Microsoft.PowerFx.Core/Lexer/TexlLexer.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (akshar2401) [reply src/libraries/Microsoft.PowerFx.Core/Lexer/TexlLexer.cs]: looks good to me. thanks for adding good test cases

**Review by bryancroteau-MSFT — APPROVED** [review]

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**